api/extenders/listeners.go

```diff
@@ -0,0 +1,31 @@
+package extenders
```

Comment:
Not a fan of this package name, any ideas?

Reply:
why not just extensions?

Reply:
Could do that (think that's what I had at first, but then it felt like it would be a directory of extensions, not just the interfaces). Would be nice to drop the whole `api` dir too, poorly named and unnecessary.

Reply:
`extensions` / `listeners` / `wrappers` all make sense. don't like wrappers, but maybe some synonym there.